src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -98,26 +100,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
       // TODO: Call scraping function to return JSON card content
     }
 
-    request.getRequestDispatcher("WEB-INF/templates/search.jsp").forward(request, response);
+    response.setContentType("text/html;");
+    response.getWriter().println(allLinks);
   }
 
   private String getLink(String id, String query) {
-    String cse_request = CSE_URL + "?key=" + API_KEY + "&cx=" + id + "&q=" + query;
-    HttpRequest linkRequest =
-        HttpRequest.newBuilder()
-            .GET()
-            .uri(URI.create(cse_request))
-            .setHeader("User-Agent", "Java 11 HttpClient Bot")
-            .build();
-    HttpResponse<String> linkResponse;
+    String url = CSE_URL + "?key=" + API_KEY + "&cx=" + id + "&q=" + query;
     try {
-      linkResponse = httpClient.send(linkRequest, HttpResponse.BodyHandlers.ofString());
-    } catch (IOException | InterruptedException ex) {
+      CloseableHttpClient httpClient = HttpClients.createDefault();
+      CloseableHttpResponse response = httpClient.execute(new HttpGet(url));
+      HttpEntity entity = response.getEntity();
+      if (entity == null) {
+        return null;
+      }
+      BufferedReader reader = new BufferedReader(new InputStreamReader(entity.getContent()));
+      StringBuilder responseBody = new StringBuilder();
+      String line;
+      while ((line = reader.readLine()) != null) {
+        responseBody.append(line);
+      }
+      JSONObject json = new JSONObject(responseBody.toString());
+      if (!json.isNull(CSE_ITEMS) && !json.getJSONArray(CSE_ITEMS).isNull(0)) {
+        JSONObject obj = json.getJSONArray(CSE_ITEMS).getJSONObject(0);
+        if (!obj.isNull(CSE_LINK)) {
+          return obj.getString(CSE_LINK);
+        }
+      }
+      return null;
+    } catch (IOException e) {
```

Comment:
a long try-catch block is not desired according to @anbenson

Reply:
Yeah :/ I certainly don't like it, but I don't see a good way around it this time short of doing manual close() which seems worse.